cli/medperf/commands/association/priority.py

```diff
@@ -0,0 +1,83 @@
+from medperf import config
+from medperf.exceptions import CleanExit, InvalidArgumentError
+
+
+class AssociationPriority:
+    @staticmethod
+    def run(
+        benchmark_uid: str, mlcube_uid: str, priority: int,
+    ):
+        """Sets priority for an association between a benchmark and an mlcube
+
+        Args:
+            benchmark_uid (str): Benchmark UID.
+            mlcube_uid (str): MLCube UID.
+            priority (int): priority value (rank). A positive integer or (-1) for least priority"
+
+        """
+
+        priority_setter = AssociationPriority(benchmark_uid, mlcube_uid, priority)
+        priority_setter.validate()
+        priority_setter.convert_priority_to_float()
+        priority_setter.update()
+
+    def __init__(self, benchmark_uid: str, mlcube_uid: str, priority: int):
+        self.benchmark_uid = benchmark_uid
+        self.mlcube_uid = mlcube_uid
+        self.priority = priority
+        self.float_priority = None
+        self.min_difference = None
+        self.priorities = []
+
+    def validate(self):
+        if self.priority < 1 and self.priority != -1:
+            raise InvalidArgumentError(
+                "Priority value must be a positive integer or (-1) for least priority"
+            )
+
+        self.assocs = config.comms.get_benchmark_model_associations(self.benchmark_uid)
+        associated_cubes = [assoc["model_mlcube"] for assoc in self.assocs]
+        if self.mlcube_uid not in associated_cubes:
+            raise InvalidArgumentError(
+                "The given mlcube is not associated with the benchmark"
+            )
+
+        cube_priority = associated_cubes.index(self.mlcube_uid) + 1
+        redundant_command = any(
+            [
+                cube_priority == len(self.assocs) and self.priority >= cube_priority,
```

Comment:
Could you provide some comments to explain these comparisons? I want to also discuss possible edge cases here but I firs need to understand what each check does.